src/__tests__/SuspenseSSR-test.tsx

```diff
@@ -0,0 +1,110 @@
+import ApolloClient from 'apollo-client';
+import { ApolloLink } from 'apollo-link';
+import { MockedResponse } from 'apollo-link-mock';
+import gql from 'graphql-tag';
+import * as React from 'react';
+import { renderToString } from 'react-dom/server';
+import { render } from 'react-testing-library';
+
+import { ApolloProvider } from '../ApolloContext';
+import { unstable_SuspenseSSR as SuspenseSSR } from '../SuspenseSSR';
+import createClient from '../__testutils__/createClient';
+import flushEffectsAndWait from '../__testutils__/flushEffectsAndWait';
+import { getMarkupFromTree } from '../getMarkupFromTree';
+import { QueryHookOptions, useQuery } from '../useQuery';
+
+const USER_QUERY = gql`
+  {
+    currentUser {
+      firstName
+    }
+  }
+`;
+
+interface UserQueryResult {
+  currentUser: { firstName: string };
+}
+
+const MOCKS: MockedResponse[] = [
+  {
+    request: { query: USER_QUERY },
+    result: { data: { currentUser: { firstName: 'James' } } },
+  },
+];
+
+function createMockClient(link?: ApolloLink) {
+  return createClient({ link, mocks: MOCKS, addTypename: false });
+}
+
+function UserDetails(props: QueryHookOptions<{}>) {
+  const { data, loading } = useQuery<UserQueryResult>(USER_QUERY, props);
+
+  return (
+    <>
+      {loading
+        ? 'Loading'
+        : !data
+        ? 'No Data'
+        : !data.currentUser
+        ? 'No Current User'
+        : data.currentUser.firstName}
+    </>
+  );
+}
+
+interface UserWrapperProps extends QueryHookOptions<{}> {
+  readonly client: ApolloClient<object>;
+}
+
+function UserDetailsWrapper({ client, ...props }: UserWrapperProps) {
+  return (
+    <ApolloProvider client={client}>
+      <SuspenseSSR fallback={<>Loading with suspense</>}>
+        <UserDetails {...props} />
+      </SuspenseSSR>
+    </ApolloProvider>
+  );
+}
+
+describe.each([[true], [false]])('SuspenseSSR with "suspend: %s"', suspend => {
+  it('not throws in react-dom', async () => {
+    const client = createMockClient();
+
+    const { container } = render(
+      <UserDetailsWrapper client={client} suspend={suspend} />
+    );
+
+    if (suspend) {
```

Comment:
I kept using `toMatchInlineSnapshot` to keep to "visual" and without manual labor 🙂